src/bindings/python/flux/job/executor.py

```diff
@@ -442,17 +442,37 @@ def shutdown(self, wait=True, *, cancel_futures=False):
             for thread in self._executor_threads:
                 thread.join()
 
-    def submit(self, jobspec):
+    def submit(self, *args, **kwargs):
         """Submit a jobspec to Flux and return a ``FluxExecutorFuture``.
 
+        Accepts the same positional and keyword arguments as ``flux.job.submit``,
+        except for the ``flux.job.submit`` function's first argument, ``flux_handle``.
+
+        :param jobspec: jobspec defining the job request
+        :type jobspec: Jobspec or its string encoding
+        :param urgency: job urgency 0 (lowest) through 31 (highest)
+            (default is 16).  Priorities 0 through 15 are restricted to
+            the instance owner.
+        :type urgency: int
+        :param waitable: allow result to be fetched with ``flux.job.wait()``
+            (default is False).  Waitable=True is restricted to the
+            instance owner.
+        :type waitable: bool
+        :param debug: enable job manager debugging events to job eventlog
+            (default is False)
+        :type debug: bool
+        :param pre_signed: jobspec argument is already signed
+            (default is False)
+        :type pre_signed: bool
```

Comment:
Is there a way to cross reference this documentation instead of copying it here? I worry that we'll forget to update this docstring when if/when we add any new parameters to `job.submit`.

Reply:
I was thinking about this. As a Python docstring that comes up from `help()` I don't think there's any way to cross-reference. In the sphinx documentation, though, it would certainly be possible to embed a link of some kind. At the moment, though, I think all the sphinx documentation is auto-generated, and I don't really know how to do it in that case.

Reply:
I don't either, but thought I'd ask. Maybe we'll come up with a solution in the future!

Reply:
(Two random thoughts)

Looks like there is this interesting `docrep` module that might help if we want to do the setup manually (and vendor another dep): https://docrep.readthedocs.io/en/latest/

Another idea would be to use Sphinx's ref capabilities so that (at least when using RTD), you can click the ref to `flux.job.submit` and be taken immediately to the right documentation.  Sadly, that doesn't help much in the IDE/auto-complete case.

Reply:
I think the Flux's Python sphinx documentation would really benefit from a little bit of custom guiding text, above and beyond the docstrings and ``automodule``---that sort of thing can really help users understand how you are supposed to combine the various parts of the interface. And once you start going down that route it wouldn't be too hard to add the refs.

If I get time I may make a PR like that.